### PR TITLE
Don't localize deletion emails sent to admins (bug #909)

### LINKED ISSUE #909: Do not localize add-on deletion emails
https://github.com/mozilla/olympia/blob/master/apps/addons/models.py#L458 sends emails when an add-on is being deleted. The property `atype` is being localized. We don't want that as it makes it very hard to search in those emails.

I'd give a shot a this myself, but I don't know why l10n happens in the first place. Any hints are appreciated.

Example:
`The following ΘΈΜΑ was deleted.`

### PR DESCRIPTION
**Pull request by magopian**
Fixes #909

### CODE REVIEW

**Comment by magopian** [comment]
Supersedes #932 (thanks @wagnerand !)

File: apps/addons/models.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from django.db import models, transaction
 from django.dispatch import receiver
 from django.db.models import Max, Q, signals as dbsignals
-from django.utils.translation import trans_real as translation
+from django.utils.translation import trans_real
```
Comment (magopian) [inline apps/addons/models.py]: Not really relevant to this change, but really, it was awfully confusing

File: apps/amo/utils.py [hunk 2]
```diff
@@ -770,11 +771,11 @@ def no_translation(lang=None):
     """
     old_lang = translation.trans_real.get_language()
     if lang:
-        translation.trans_real.activate(lang)
+        tower.activate(lang)
```
Comment (magopian) [inline apps/amo/utils.py]: Tower is doing clever things under the hood, so we need to use it to "really" (de)activate a locale.

**Comment by wagnerand** [comment]
Thank you @magopian!

**Comment by EnTeQuAk** [comment]
Besides travis failing I would kind of like having the whole code-branch in the `with no_translation()` statement rather then having it being called twice. Makes it a bit more obvious to reason about.

File: apps/addons/models.py [hunk 3]
```diff
@@ -485,7 +488,9 @@ def delete(self, msg='', reason=''):
             REASON GIVEN BY USER FOR DELETION: %(reason)s
             """ % context
             log.debug('Sending delete email for %(atype)s %(id)s' % context)
-            subject = 'Deleting %(atype)s %(slug)s (%(id)d)' % context
+            # Don't localize email to admins, use 'en-US' always.
+            with no_translation():
```
Comment (magopian) [inline apps/addons/models.py]: actually this one isn't needed, it's only the type that is localized, and we already "un-localized" it above ;)

**Comment by dbaron** [comment]
Hey, this created bogus comments in https://bugzilla.mozilla.org/show_bug.cgi?id=909 .  Could you try to avoid doing that?  (I'm not sure how the github robot decides when to comment in a bug, but it wasn't the right thing here.)

**Comment by wagnerand** [comment]
The github robot took thegithub  issue number (#909) and interpreted as a bugzilla bug number (bug 909). We should probably avoid using the term "bug" for github issues or adapt the bot.

Sorry for that.

**Pull request merged**